File: Web/WeatherApp.API/Models/WeatherModel.cs
using System;
using WeatherApp.API.Models.WeatherDetails;

namespace WeatherApp.API.Models
{
    public class WeatherModel
    {
        public Weather[] Weather { get; set; }
        public Main Main { get; set; }
        public Wind Wind { get; set; }
        public Clouds Clouds { get; set; }
        public Sys Sys { get; set; }
        public int Visibility { get; set; }
        public double Dt { get; set; }

        // City id
        public int Id { get; set; }

        //City name
        public string Name { get; set; }

    }
}

File: Client/WeatherApp-Client/src/app/shared/models/weather.model.ts
import { Main } from "./main";
import { Weather } from "./Weather";
import { Wind } from "./Wind";
import { Clouds } from "./Clouds";
import { Sys } from "./Sys";

export class WeatherModel {
  main: Main;
  weather: Weather[];
  wind: Wind;
  clouds: Clouds;
  sys: Sys;
  visibility: number;
  dt: number;
  id: number;
  name: string;
}
